### PR TITLE
ethereum: new implementation check

### PR DESCRIPTION
**Pull request by evan-gray**
Recently ran into a case where it's possible to mistakenly set an implementation for a different bridge as the target of an upgrade. This change should ensure that bridges are always updated to their same implementation.

---
This change is [Reviewable](https://reviewable.io/reviews/certusone/wormhole/1017)

### CODE REVIEW

**Comment by evan-gray** [comment]
Could use a few tests, but wanted to collect feedback.

**Comment by kcsongor** [comment]
Really good idea. I think we could get away with a single interface, something like `IWormholeModule`, then external tooling could query the module name in a uniform way.

**Review by bruce-riley — APPROVED** [review]
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/certusone/wormhole/1017)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @calebrate and @hendrikhofstadt)

File: ethereum/contracts/Governance.sol [hunk 1]
```diff
@@ -10,13 +10,21 @@ import "./Setters.sol";
...
+}
+
 abstract contract Governance is GovernanceStructs, Messages, Setters, ERC1967Upgrade {
     event ContractUpgraded(address indexed oldContract, address indexed newContract);
     event GuardianSetAdded(uint32 indexed index);
 
     // "Core" (left padded)
     bytes32 constant module = 0x00000000000000000000000000000000000000000000000000000000436f7265;
 
+    function getModule() public pure returns (bytes32) {
```
Comment (kcsongor) [inline ethereum/contracts/Governance.sol:24]: It looks like none of these contracts actually _implement_ the associated interface (in the sense of nominal typing, `... is CoreBridgeModule`). Is that intentional?

**Review by bruce-riley — APPROVED** [review]
> Reviewed 4 of 4 files at r2, 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/certusone/wormhole/1017)* status: all files reviewed, 1 unresolved discussion (waiting on @calebrate, @evan-gray, and @hendrikhofstadt)

**Comment by evan-gray** [comment]
Closing for now - mixed responses on size increase versus safety and better tooling with ethereum/verify

**Pull request closed without merging**